frontend/src/basic/Main.tsx

```diff
@@ -35,39 +30,33 @@ const MainBox = styled(Box)<MainBoxProps>((props) => ({
 
 const Main: React.FC = () => {
   const { t } = useTranslation();
-  const {
-    settings,
-    robot,
-    setRobot,
-    baseUrl,
-    order,
-    page,
-    slideDirection,
-    setOpen,
-    windowSize,
-    navbarHeight,
-  } = useContext<UseAppStoreType>(AppContext);
+  const { settings, page, slideDirection, setOpen, windowSize, navbarHeight, hostUrl, origin } =
+    useContext<UseAppStoreType>(AppContext);
+  const { federation, sortedCoordinators } = useContext<UseFederationStoreType>(FederationContext);
+  const { garage } = useContext<UseGarageStoreType>(GarageContext);
+  const [avatarBaseUrl, setAvatarBaseUrl] = useState<string>(hostUrl);
+
+  useEffect(() => {
+    setAvatarBaseUrl(federation.getCoordinator(sortedCoordinators[0]).getBaseUrl());
+  }, [settings.network, settings.selfhostedClient, federation, sortedCoordinators]);
 
   return (
     <Router>
       <RobotAvatar
         style={{ display: 'none' }}
-        nickname={robot.nickname}
-        baseUrl={baseUrl}
+        nickname={garage.getRobot().nickname}
+        baseUrl={federation.getCoordinator(sortedCoordinators[0]).getBaseUrl()}
```

Comment:
@KoalaSat my intuition is pointing that the issue with the robot avatar loading might be here`federation.getCoordinator(sortedCoordinators[0]).getBaseUrl()` . We have the first index of sortedCoordinators hardcoded`sortedCoordinators[0]`, however, the first coordinator might not have create the robot. It feels like we might want to track in the the Garage what coordinators have replied to the `/api/robot` request successfully, and then only try to fetch the avatar from a coordinator who did create the robot. What do you think?